src/Tasks/SystemState.cs

```diff
@@ -607,21 +615,19 @@ private string[] GetDirectories(string path, string pattern)
         /// <returns>True if the file exists.</returns>
         private bool FileExists(string path)
         {
-            /////////////////////////////////////////////////////////////////////////////////////////////
-            // FIRST -- Look in the primary cache for this path.
```

Comment:
These comments are overwrought but I think having _some_ comment would be reasonable. Maybe just

```
// Cache miss; fall back to actual check and populate cache with result
```